luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -106,25 +119,40 @@ function gadget:UnitPreDamaged(unitID, unitDefID, unitTeam, damage, paralyzer, w
 end
 
 function gadget:UnitLoaded(unitID, unitDefID, unitTeam, transportID, transportTeam)
-	transportingUnits[transportID] = transportingUnits[transportID] or {}
-	transportingUnits[transportID][unitID] = true
+	transportedUnits[unitID] = true
 end
 
 function gadget:UnitUnloaded(unitID, unitDefID, unitTeam, transportID, transportTeam)
-	transportingUnits[transportID][unitID] = nil
+	transportedUnits[unitID] = nil
 end
 
 function gadget:UnitDestroyed(unitID)
-	transportingUnits[unitID] = nil
+	transportedUnits[unitID] = nil
+	unitInertiaCheckFlags[unitID] = nil
 end
 
 function gadget:GameFrame(frame)
-	if frame % 5 == 3 then
-		for unitID, data in pairs (overImpulseCooldowns) do
-			if data.expireframe < frame  then --this is done to prevent a unit from being impulsed multiple times in a compounding fashion in a performant way.
-				overImpulseCooldowns[unitID] = nil
+	for unitID, expirationFrame in pairs(unitInertiaCheckFlags) do
+		if not transportedUnits[unitID] then
+			local velocityCap = 3
```

Comment:
This value should be grouped with the other config values at the start of the file

Reply:
done